specifyweb/frontend/js_src/lib/components/TreeView/Tree.tsx

```diff
@@ -151,12 +152,18 @@ export function Tree<SCHEMA extends AnyTree>({
                 role="columnheader"
               >
                 {index === 0 ? (
-                  <Button.Icon
-                    aria-pressed={isEditingRanks}
-                    icon="pencil"
-                    title={treeText.editRanks()}
-                    onClick={handleToggleEditingRanks}
-                  />
+                  <>
+                    <Button.Icon
+                      aria-pressed={isEditingRanks}
+                      icon="pencil"
+                      title={treeText.editRanks()}
+                      onClick={handleToggleEditingRanks}
+                    />
+                    <AddRank
```

Comment:
before displaying `AddRank`, you should check for `create` permission for the `treeDefinitionItems[0]._tableName` table